### PR TITLE
6328 fix http path normalization 1.10

### PR DESCRIPTION
**Pull request by JoshVanL**
See #6332

### CODE REVIEW

File: docs/release_notes/v1.10.7.md
Comment (ItalyPaleAle) [inline docs/release_notes/v1.10.7.md:25]:
```suggestion
When using the Dapr sidecar HTTP APIs, some HTTP URLs where not being routed to
```

File: docs/release_notes/v1.10.7.md
Comment (ItalyPaleAle) [inline docs/release_notes/v1.10.7.md:26]:
```suggestion
the correct handler. For example, when prefixing the URL path with duplicated
```

File: docs/release_notes/v1.10.7.md
Comment (ItalyPaleAle) [inline docs/release_notes/v1.10.7.md:31]: Please indicate the version this issue started. I suppose 1.10.0?

File: docs/release_notes/v1.10.7.md
Comment (ItalyPaleAle) [inline docs/release_notes/v1.10.7.md:36]: I would keep it less specific.

> The HTTP router inside Dapr was "normalizing" the request's HTTP path only after evaluating the route, causing routing errors with un-normalized paths.

File: docs/release_notes/v1.10.7.md
Comment (ItalyPaleAle) [inline docs/release_notes/v1.10.7.md:43]: Keep it generic, since this won't mean much for someone who's not a Dapr core developer.

> We changed the router used by the Dapr HTTP server to "normalize" HTTP paths before matching the route.

File: docs/release_notes/v1.10.7.md
Comment (JoshVanL) [reply docs/release_notes/v1.10.7.md:31]: Yeah, that looks to be the case from my testing

File: docs/release_notes/v1.10.7.md
Comment (JoshVanL) [reply docs/release_notes/v1.10.7.md:36]: :+1: 

Btw, *normalization* is a well defined term that doesn't need quoting. https://datatracker.ietf.org/doc/html/rfc3986#section-6

**Comment by JoshVanL** [comment]
@artursouza @ItalyPaleAle PTAL

**Review by ItalyPaleAle — APPROVED** [review]

**Comment by artursouza** [comment]
I cherry-picked my E2E test change to make sure this regression does not come back. I will merge after the E2E tests pass.

**Review by artursouza — APPROVED** [review]

**Pull request merged**
